cmd/opampsupervisor/supervisor/opamplogger.go

```diff
@@ -0,0 +1,27 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package supervisor
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opamp-go/client/types"
+	"go.uber.org/zap"
+)
+
+// opAMPLogger adapts the Supervisor's zap.Logger instance so it
+// can be used by the OpAMP client.
```

Comment:
I don't think it does that. It adds a couple convenience methods that allows to compose strings using Sprintf. I am not fond of this, as zap fields are meant to reduce the cost of concatenation in case the log doesn't happen, but it ends up being a matter of personal preference. Not blocking this PR.

Reply:
I agree it's not an ideal solution, it was just a way to deal with the fact that Zap wants `zap.Field` and the OpAMP client logging interface accepts `any`.

You made me realize I completely forgot about `zap.SugaredLogger`, so I've switched to just passing the OpAMP client the sugared logger directly, which obviates the need for this adapter struct.